mantle/util/common.go

```diff
@@ -120,3 +121,43 @@ func RunCmdTimeout(timeout time.Duration, cmd string, args ...string) error {
 		return fmt.Errorf("%s timed out after %s", cmd, timeout)
 	}
 }
+
+// ParseDiskSpec converts a disk specification into a Disk. The format is:
+// <size>[:<opt1>,<opt2>,...], like ["5G:channel=nvme"]
+func ParseDiskSpec(spec string) (string, map[string]string, error) {
+	diskmap := map[string]string{}
+	split := strings.Split(spec, ":")
+	if split[0] == "" {
+		return "", nil, fmt.Errorf("invalid size %s", spec)
+	}
+	var disksize string
+	if len(split) == 1 {
+		disksize = split[0]
```

Comment:
The idea I had was to paese the disk size into an integer in this function. To start, we can support just gigabytes (i.e. optionally strip a `G` at the end of the string, then parse as a uint; let it fail if other characters are present). WDYT?

Reply:
SGTM

Reply:
Consider about qemu, https://github.com/coreos/coreos-assembler/blob/main/mantle/platform/qemu.go#L89, which needs string to pass to [qemu-img](https://github.com/coreos/coreos-assembler/blob/main/mantle/platform/qemu.go#L1049-#L1052), and support "K", "M", "T", I might misunderstood.
```
type Disk struct {
	Size          string   // disk image size in bytes, optional suffixes "K", "M", "G", "T" allowed.
```

Reply:
Right yeah, so for QEMU, the code would reconvert to a string, e.g. `fmt.Sprintf("%dG", size)`.